适配1.19

MATRIX-feather opened the pull request:
不清楚要不要继续尝试兼容基岩版，所以相关功能暂时没考虑。

重生规则暂时禁用，因为还没确定要不要重新设计新机制。

*开完pr才发现分支名少个dev*

MATRIX-feather committed:
rm: 移除不需要的数据包和文件

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
misc: 更新Teams

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
misc: 更新nifeather_extras到1.19
misc: 暂时禁用3lives
	* 重生机制可能需要调整

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
misc: 禁用nifeather_extras:village维度
misc: 更新README

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

ALKALiKong233 commented:
@MATRIX-feather 建议对基岩版不用再做特别的适配，跟着 java 版凑合用就完了（）。另外现在考虑到 java 版应该能普及一些了所以我觉得基岩版不用再放那么重要了

MATRIX-feather commented:
好，那我这两天找时间清理一下数据包和资源包里面现有的兼容方案

MATRIX-feather committed:
misc: 清理部分基岩版兼容
rm: 移除多余的文件
misc: 更改 nifeather_extras:lmb/house/0 的结构内容

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
fix: pixkaxe -> pickaxe

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
fix: 使用flat_world_preset.minecraft.overworld作为显示"主世界"的本地化键名

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
misc: 将nifeather_extras:chests从nifeather_extras:base中移除
	* Spigot 1.19不知道为什么只有创造模式能放带LootTable的箱子，暂时移除

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
add: 添加GM4数据包

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
misc: 更新Teams

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
add: 添加custom crafters

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
Revert "add: 添加custom crafters"

This reverts commit 736b2bab6e7fb8072372efc7b1ce3e5b1f89e691.

MATRIX-feather committed:
Revert "misc: 将nifeather_extras:chests从nifeather_extras:base中移除"
	* Paper修复了Spigot的问题

This reverts commit 7a8ac4a74c7c8eea1b7d021f2621e008e70a9813.

MATRIX-feather committed:
misc: 调整nifeather_extras:base掉落表中potions/base的概率

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
misc: 更改limbo环境为the_nether
add: 添加darkness作为limbo的debuff药效

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
rm: 移除针对indendium的史莱姆修复

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
misc: 延迟limbo的凋零效果

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
fix: 延长darkness持续时间
	* 忘记了limbo的loop_boot是每两秒运行一次...

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
enchance: 使最大生命值随玩家盔甲进度而改变
misc: 移动部分函数和进度的位置
enchance: 重新启用nifeather_3lives

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
new: 实现新的旁观者重生策略

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
misc: 将nife_lives更名为n3l_lives
misc: 文案调整

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
misc: 确保玩家死亡时n3l_lives的值总是大于等于0

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
fix: 修复reset_n3l_data没有完全重置某一玩家的N3L数据

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
misc: 重命名标签(3livesinit -> n3l_init_done)

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
misc: 重命名标签(主要将attribute中的"nife_"替换为"n3l_")

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
misc: 一些调整

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
new: 添加重生倒计时显示
misc: 将默认等待时间延长至3小时
rm: 移除不需要的东西

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
misc: 调整词条道具的获得方式
	* 词条道具再也不能通过钓鱼获得了
	* 现在将通过在锻造台内锻造图腾和金苹果获得
misc: 文案调整

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
misc: 使词条道具尽可能无法堆叠

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
misc: 调换词条道具的配方顺序
	* 这样看上去会更合理些（

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
enchance: 优化阻止堆叠词条道具的方式
	* 之前是从世界里随机取一个非玩家实体的UUID，在实体少的情况下容易获取重复的RNG数值
	* 新方案是生成一个新UUID + 设置序号，可以极大程度地降低堆叠概率
	* 而且还可以统计总共合成了多少个词条道具（

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
misc: 限制知识之书的检查范围

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
fix: 对处于治疗状态下的玩家使用不同的加入文案

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
fix: ...

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
fix: 修复时间显示x2

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
misc: 调整函数运行顺序

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
fix: 修复治疗结束时词条道具有概率同时触发的问题
misc: 调整n3l_curing标签的添加时间
	* 现在会在玩家重生后打上，这样也可以优化相关消息

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>

MATRIX-feather committed:
misc: 调整初始最大生命到15

Signed-off-by: MATRIX-feather <midnightcarnival@outlook.com>
